Test cases for Property Model

Jeshan7 opened the pull request:
## Why?
 - To add test cases for the property model.
## This change addresses the need by:
 - Added a file in test/data/model/property_test.exs

[delivers ##163144092]

## Checklist
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] I have read [CONTRIBUTING.md][contributing].
- [ ] My code follows the [style guidelines][contributing] of this project.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] My code does not generate any (new) [`credo`][credo] and compile-time warnings.
- [ ] I have updated the documentation wherever necessary.
- [ ] I have added tests to cover my changes.

[contributing]: https://github.com/aviabird/snitch/CONTRIBUTING.md
[credo]: https://github.com/rrrene/credo

Jeshan7 committed:
[#163144092] Added test cases for Property model

ashish173 commented on apps/snitch_core/test/data/model/property_test.exs:
```diff
@@ -0,0 +1,70 @@
+  }
+
+  describe "create/2" do
+    test "create sucessfully" do
```
@Jeshan7 the way you are writing describe and test. It would read "create/2 create sucessfully". 

This doesn't read nicely. I would suggest you look at other test cases.

Jeshan7 replied:
Okay

Jeshan7 committed:
[#163144092] Made changes in describe and test

ashish173 commented on apps/snitch_core/test/data/model/property_test.exs:
```diff
@@ -0,0 +1,70 @@
+      assert {:ok, _} = Property.create(@params)
+    end
+
+    test "creation fails for duplicate name" do
```
same here

ashish173 commented on apps/snitch_core/test/data/model/property_test.exs:
```diff
@@ -0,0 +1,70 @@
+  end
+
+  describe "update/2" do
+    test "update with " do
```
here also

Jeshan7 replied:
I have made the changes

SagarKarwande commented on apps/snitch_core/test/data/model/property_test.exs:
```diff
@@ -0,0 +1,70 @@
+  alias Snitch.Data.Schema.Property, as: PropertySchema
+
+  @params %{
+    name: "Sapphire Radeon",
```
`name` is the product attribute and `display name` the name that will be displayed on the frontstore.

I think here we are can use following 
```elixir
%{
   name: "processor count",
   display_name: "Processors Count"
}
```

SagarKarwande commented on apps/snitch_core/test/data/model/property_test.exs:
```diff
@@ -0,0 +1,70 @@
+    end
+  end
+
+    test "get property" do
```
Put this test in describe block

SagarKarwande commented on apps/snitch_core/test/data/model/property_test.exs:
```diff
@@ -0,0 +1,70 @@
+    end
+
+  describe "delete/1" do
+    test "a property" do
```
Name the test for which it is intended to perform

Jeshan7 replied:
I didn't understand this change

SagarKarwande replied:
Here we are testing if we can delete a property using `delete` method.
The test name can be `Successfully delete a property`.

This will give the reader the test intent.

Jeshan7 replied:
Okay

SagarKarwande commented on apps/snitch_core/test/data/model/property_test.exs:
```diff
@@ -0,0 +1,76 @@
+    end
+  end
+
+
```
Leave single line between describe blocks.

Jeshan7 committed:
[#163144092]
    This change addresses the need by:
     - Changed :name and :display_name attributes in params
     - Put tests "get/1" and "get_all/0" inside describe function
     - Changed the value of describe "delete/1" test

SagarKarwande commented on apps/snitch_core/test/data/model/property_test.exs:
```diff
@@ -0,0 +1,74 @@
+      %{id: expected_id} = property
+      updates = %{name: "Amazon TV"}
+      assert {:ok, %{id: received_id}} = Property.update(property, updates)
+      assert expected_id == received_id
```
I think we should assert the updated value. The id won't change.

Jeshan7 committed:
[#163144092]
   Before Commit
    - Test update/2 was only checking the new and old id of property that were inserted.
   After Commit
    - Test update/2 now check whether the new and old ids of property are same and the updated value is different from old value.
